src/libasr/asr_verify.cpp

```diff
@@ -420,6 +420,14 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
     }
 
     void visit_Function(const Function_t &x) {
+        ASR::FunctionType_t* v_func_type = ASR::down_cast<ASR::FunctionType_t>(x.m_function_signature);
+        if (v_func_type->m_abi == abiType::Interactive) {
+            require(x.n_body == 0,
+            "The Function::n_body should be 0 if abi set to Interactive");
+            require(x.m_body == nullptr,
+            "The Function::m_body should be null if abi set to Interactive");
+            return;
```

Comment:
I think this should be without the return:
```suggestion
```
Then it would be good, it will make the ASR verify stronger.

Reply:
>I think this should be without the return:

Yes, it should be removed. @Vipul-Cariappa  since you have issues with the dependencies for interactive, how about we make just the dependency check to be optional? Or better we make the dependency check to be having no dependencies?

And we can also update the `mark_all_variables_external()` here https://github.com/lfortran/lfortran/blob/d0882b59a1002efccc8532ee235f859271ef658e/src/libasr/asr_scopes.cpp#L39-L45 to remove the dependencies (along with the body).

Reply:
You can remove the dependency check in a Draft PR to see if it fixes issues.

However, to merge into master, we need to keep all these checks, they are very important. Instead, we need to just remove all dependencies when we are removing the body. To be exact, we should keep dependencies if they are used in declarations, but removing all of them might get things working pretty well and we can refine it later. The ASR checks I think should all stay, with the addition of the above ones for interactive.

Reply:
> To be exact, we should keep dependencies if they are used in declarations

Do we know of any case where we will require the dependencies of a function during its declaration, but it does not require the body of the function? Also a thing to note is that we would have changed the ABI to interactive.

---

I am not very familiar with dependency, can someone please explain the requirement of the dependency, and how it is used in later parts of the compiling stages? I assume we verify the dependencies by going through the function body and checking if it calls other functions and then checking if that function is in the dependency list.

Reply:
Yes, for example:
```fortran
interface
  subroutine s(n, x)
  integer, intent(in) :: n
  real, intent(inout) :: x(f(n))
  end subroutine
interface
```
Here `s` depends on the pure function `f`.